### PR TITLE
Add support for "rewritten urls" in module detection

### PR DESCRIPTION
**Pull request by dweeves**
This is a fix for issue #1023 & duplicated #1093

The code adds a new method to get module information from url.
This method is first used on the actual symfony request pathinfo
if no valid module is found, then the same method is applied on the real 'PATH_INFO' (which will be the rewritten on) .

The last commit is now supporting urls without trailiing slash (/register)

### CODE REVIEW

**Comment by dweeves** [comment]
Any news on final approbation ?

**Comment by tvdijen** [comment]
Hi @dweeves !

I've asked Jaime again to take a look at this PR. I know there are several version of 'routing improvement' lying around and we need to pick a strategy here. A finalized version of the routing-feature will be part of a  v1.18-release, hopefully somewhere in Q4 this year depending on resources.

File: lib/SimpleSAML/Module.php [hunk 1]
```diff
@@ -126,31 +149,20 @@ public static function process(Request $request = null)
...
-         *'.php'-ending.
-         */
-        unset($_SERVER['PATH_INFO']);
-
-        $modEnd = strpos($url, '/', 1);
-        if ($modEnd === false) {
-            // the path must always be on the form /module/
-            throw new Error\NotFound('The URL must at least contain a module name followed by a slash.');
+        if ($basepathinfo === '/') {
+            throw new Error\NotFound('No PATH_INFO to module.php');
```
Comment (jaimeperez) [inline lib/SimpleSAML/Module.php]: This error would be shown to the end user, but won't help much, I'm afraid. Should we have an error message similar to the old one?

Comment (dweeves) [reply lib/SimpleSAML/Module.php]: I agree that the error message is not giving much valuable information, so using the old one is a good option.

About the use of  $_SERVER['path_info'] , i didn't find a way to have it through Symfony request. i had a look at the code inside symfony and the thing is they are trying to be "too smart" about request parsing, providing a "too high" abstraction level above the initial request. so loosing information with no way to get it back from the original request. So the only way i found was to ask the original request details directly through $_SERVER.

Comment (jaimeperez) [reply lib/SimpleSAML/Module.php]: Uhm, I see...

Do you think this is actually wrong behaviour in Symfony? Should we report it upstream for them to consider a change (regardless of whether we work around it or not)?

Comment (dweeves) [reply lib/SimpleSAML/Module.php]: I think this is a behaviour that is not ideal for the purpose we want to achieve (ie, detecting which module is targeted by the  rewritten url) , i cannot say it's a "bad" behaviour from symfony, it's just a choice that was made that is making the request object not well suited for low level url access but very interesting to process request content & structure.

**Pull request closed without merging**